qiita_db/artifact.py

```diff
@@ -509,10 +509,9 @@ def delete(cls, artifact_id):
                 raise qdb.exceptions.QiitaDBArtifactDeletionError(
                     artifact_id, "it is public")
 
-            children = instance.children
-            children_ids = [c.id for c in children]
-            all_ids = tuple(children_ids + [artifact_id])
-            all_artifacts = children + [instance]
+            all_artifacts = list(instance.descendants.nodes())
+            all_artifacts.reverse()
```

Comment:
Approved and merged, but if you can comment, it would be appreciated.

Reply:
I actually thought the same but turns out that it matters for the delete below where if not in the right order it will raise an error as you are deleting a parent that still has a child ...